internal/bminventory/inventory.go

```diff
@@ -773,17 +765,17 @@ func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params inst
 
 func (b *bareMetalInventory) DownloadClusterISOHeaders(ctx context.Context, params installer.DownloadClusterISOHeadersParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
-	var cluster common.Cluster
+	var infraEnv common.InfraEnv
 
-	if err := b.db.First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
-		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+	if err := b.db.First(&infraEnv, "id = ?", params.ClusterID).Error; err != nil {
```

Comment:
maybe it should be a helper function? or maybe use `GetInfraEnvFromDB`?

Reply:
fixed